### ISSUE TITLE
vcard encoding error: KeyError: 'ENCODING'

### ISSUE THREAD

**Issue by pbiering** [eventable/vobject#48]
Found in case of QUOTED-PRINTABLE vcards:

```
2016-10-16 20:01:22,601 - [7f3062f17700] ERROR: Exception details:
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/radicale/storage.py", line 760, in get
    item = vobject.readOne(text)
  File "/usr/lib/python3.4/site-packages/vobject/base.py", line 1130, in readOne
    allowQP))
  File "/usr/lib/python3.4/site-packages/vobject/base.py", line 1074, in readComponents
    vline = textLineToContentLine(line, n)
  File "/usr/lib/python3.4/site-packages/vobject/base.py", line 913, in textLineToContentLine
    'lineNumber' : n})
  File "/usr/lib/python3.4/site-packages/vobject/base.py", line 337, in __init__
    self.value = codecs.decode(self.value.encode("utf-8"), "quoted-printable").decode(self.params['ENCODING'])
KeyError: 'ENCODING'
```

Example:
`
LABEL;HOME;PREF;ENCODING=QUOTED-PRINTABLE:ABCstr. 22=0D=0A=0D=0A12345  Ort`

There is a bug in the coding sequence:

```
        qp = False
        if 'ENCODING' in self.params:
            if 'QUOTED-PRINTABLE' in self.params['ENCODING']:
                qp = True
                self.params['ENCODING'].remove('QUOTED-PRINTABLE')
                if 0==len(self.params['ENCODING']):
                    del self.params['ENCODING']
        if 'QUOTED-PRINTABLE' in self.singletonparams:
            qp = True
            self.singletonparams.remove('QUOTED-PRINTABLE')
        if qp:
            self.value = codecs.decode(self.value.encode("utf-8"), "quoted-printable").decode(self.params['ENCODING'])
```

"If qp" is reached in case of QUOTED-PRINTABLE, self.params['ENCODING'] is not always filled

**Comment by jelmer** [comment]
See also #40

**Comment by wpercy** [comment]
resolved by #49
